src/lambda_handler.py

```diff
@@ -109,19 +253,79 @@ def handler(event, context) -> dict[str, Optional[str]]:
         old_log_level = logging.getLogger().getEffectiveLevel()
         logging.getLogger().setLevel(new_log_level)
 
-    task_name = f"task_{event.get('task')}"
-    task = globals().get(task_name, task_default)
+    # Set up the SSM client if necessary
+    if ssm_client is None:
+        ssm_client = boto3_client("ssm")
+
+    mongodb_uri_elements: List[Tuple[str, Optional[str]]] = []
+
+    # This only runs from a CloudWatch scheduled event invocation
+    trigger_source: Optional[str]
+    trigger_type: Optional[str]
+    if (
+        (trigger_source := event.get("source")) is None
+        or trigger_source != "aws.events"
+    ) or (
+        (trigger_type := event.get("detail-type")) is None
+        or trigger_type != "Scheduled Event"
+    ):
+        logging.error("Invalid invocation event.")
+        return
+
+    for var in [
+        "ssm_db_user",
+        "ssm_db_pass",
+        "ssm_db_host",
+        "ssm_db_port",
+        "ssm_db_authdb",
+    ]:
+        mongodb_uri_elements.append((var, os.environ.get(var)))
 
-    result: dict[str, Any]
-    if not callable(task):
-        logging.error("Provided task is not a callable.")
-        logging.error(task)
-        result = task_default(event)
+    # Check that we have all of the required variables
+    if missing_variables := [k for k, v in mongodb_uri_elements if v is None]:
+        logging.error("Missing required variables: %s", ",".join(missing_variables))
+        return
+
+    # Determine the database where the CVE data will be inserted
+    write_db = get_ssm_parameter(os.environ.get("ssm_db_writedb", "ssm_db_authdb"))
+
+    # Determine if a non-default CVEs JSON URL is being used
+    nvd_urls = generate_urls()
+
+    json_url = os.environ.get("json_url")
+    if json_url is None:
+        cve_json_urls = nvd_urls
     else:
-        result = task(event)
-    response.update(result)
+        cve_json_urls = [json_url]
+
+    # Determine if a non-default collection is being used
+    db_collection = os.environ.get("target_collection")
```

Comment:
Since this is stored in SSM Parameter Store we should name it accordingly for consistency.

```suggestion
    db_collection = os.environ.get("ssm_db_target_collection")
```